applications/vanilla/models/class.discussionmodel.php

```diff
@@ -1147,7 +1147,12 @@ public function calculate(&$discussion) {
             if ($category && !is_null($category['DateMarkedRead'])) {
                 // If the category was marked explicitly read at some point, see if that applies here
                 $discussion->DateLastViewed = self::maxDate($discussion->DateLastViewed, $category['DateMarkedRead']);
+                // If the discussion was created after the category was marked read and hasn't been viewed, reset DateLastViewed to null
+                if ($category['DateMarkedRead'] <= $discussion->DateInserted && is_null($discussion->CountCommentWatch)) {
+                    $discussion->DateLastViewed = null;
```

Comment:
It looks like we're setting `$discussion->DateLastViewed`, a few lines up, and then potentially immediately resetting it here. Can we alter this logic so we only need to set the value once?

Reply:
I didn't like that either, but couldn't think of a better solution. I'm going to think harder.

Reply:
Think I figured it out.